smithy-linters/src/main/java/software/amazon/smithy/linters/ShouldHaveUsedTimestampValidator.java

```diff
@@ -160,31 +172,43 @@ private List<ValidationEvent> validateUnion(
 
     private Stream<ValidationEvent> validateTargetShape(
             String name,
-            MemberShape target,
+            MemberShape memberShape,
             Model model,
             List<Pattern> patterns
     ) {
-        return OptionalUtils.stream(model.getShape(target.getTarget())
-                .flatMap(shape -> validateName(name, shape.getType(), target, patterns)));
+        return OptionalUtils.stream(model.getShape(memberShape.getTarget())
+                .flatMap(targetShape -> validateName(name, targetShape, memberShape, patterns, model)));
     }
 
     private List<ValidationEvent> validateSimpleShape(
             Shape shape,
             List<Pattern> patterns
     ) {
-        return validateName(shape.getId().getName(), shape.getType(), shape, patterns)
-                .map(ListUtils::of)
-                .orElse(ListUtils.of());
+        String name = shape.getId().getName();
+
+        return patterns
+                .stream()
+                .filter(pattern -> pattern.matcher(name).matches())
+                .map(matcher -> buildEvent(shape, name, shape.getType()))
+                .collect(Collectors.toList());
     }
 
     private Optional<ValidationEvent> validateName(
             String name,
-            ShapeType type,
+            Shape targetShape,
             Shape context,
-            List<Pattern> patterns
+            List<Pattern> patterns,
+            Model model
     ) {
+        ShapeType type = targetShape.getType();
         if (type == ShapeType.TIMESTAMP) {
             return Optional.empty();
+        } else if (type == ShapeType.ENUM) {
+            return Optional.empty();
+        } else if (type == ShapeType.STRUCTURE || type == ShapeType.LIST) {
```

Comment:
Do we also need to check maps?

Reply:
Could we add that in this PR or do you feel that is too out of scope?

Reply:
Well at least from what we've seen in AWS services, maps are not a cause of false positives the same way list members and structure members have been. I can add it if you want, but I don't think it would add much value tbh

Reply:
Fair enough. Let's consider it out of scope for this change.